diff --git a/relation.module b/relation.module
index 7792961..3aa370b 100644
--- a/relation.module
+++ b/relation.module
@@ -622,9 +622,7 @@ function relation_save($relation, $account = NULL) {
     field_attach_validate('relation', $relation);
   }
   catch (FieldValidationException $e) {
-    watchdog_exception('relation', $e, t('Field validation failed.'), NULL,
-      WATCHDOG_ERROR);
-    return FALSE;
+    watchdog('relation', 'Field validation on relation failed', array(), WATCHDOG_ERROR);
   }
   // Determine if we will be inserting a new relation.
   if (!isset($relation->is_new)) {
